src/features/vote/RevealPhase.tsx

```diff
@@ -0,0 +1,290 @@
+  refetchEncryptedVotes,
+}) => {
+  const [tableValues, setTableValues] = useState<TableValue[]>([]);
+  const [canReveal, setCanReveal] = useState(false);
+  const {
+    state: { network, signer, provider },
+  } = useContext(OnboardContext);
+
+  const { votingContract, designatedVotingContract } = useVotingContract(
+    signer,
+    isConnected,
+    network,
+    votingAddress,
+    hotAddress
+  );
+
+  useEffect(() => {
+    if (encryptedVotes.length) {
+      if (revealedVotes.length) {
+        revealedVotes.forEach((el) => {
+          const findRevealedVote = encryptedVotes.find(
+            (x) =>
+              x.identifier === el.identifier &&
+              x.ancillaryData === el.ancillaryData &&
+              x.time === el.time
+          );
+          // If there are no revealed votes and some encrypted votes, set can reveal to true.
+          if (!findRevealedVote) setCanReveal(true);
+        });
+      } else {
+        setCanReveal(true);
+      }
+    } else {
+      setCanReveal(false);
+    }
+  }, [encryptedVotes, revealedVotes]);
+
+  // Take activeRequests and encryptedVotes and convert them into tableViews
+  useEffect(() => {
+    // Check if the user has voted in this round.
+    if (activeRequests.length && !encryptedVotes.length) {
+      const tv: TableValue[] = activeRequests.map((el) => {
+        return {
+          ancillaryData: el.ancillaryData,
+          vote: "-",
+          identifier: el.identifier,
+          revealed: false,
+          ancHex: el.idenHex,
+        };
+      });
+
+      setTableValues(tv);
+    }
+    if (activeRequests.length && encryptedVotes.length) {
+      const tv = [] as TableValue[];
+      activeRequests.forEach((el) => {
+        const datum = {} as TableValue;
+        datum.ancillaryData = el.ancillaryData;
+        datum.identifier = el.identifier;
+        let vote = "-";
+        // I believe latest events are on bottom. requires testing.
+        const latestVotesFirst = [...encryptedVotes].reverse();
+        const findVote = latestVotesFirst.find(
+          (x) =>
+            x.identifier === el.identifier &&
+            x.ancillaryData === el.ancHex &&
+            x.time === el.time
+        );
+
+        if (findVote) {
+          datum.vote = ethers.utils.formatEther(findVote.price);
+          if (el.identifier.includes("Admin")) {
+            if (datum.vote === "1" || datum.vote === "1.0") datum.vote = "Yes";
+            if (datum.vote === "0" || datum.vote === "0.0") datum.vote = "No";
+          }
+        } else {
+          datum.vote = vote;
+        }
+        const findReveal = revealedVotes.find(
+          (x) =>
+            x.identifier === el.identifier &&
+            x.ancillaryData === el.ancHex &&
+            x.time === el.time
+        );
+        if (findReveal) {
+          datum.revealed = true;
+        } else {
+          datum.revealed = false;
+        }
+
+        tv.push(datum);
+      });
+      setTableValues(tv);
```

Comment:
can we also save revealableVotes here, and remove the logic from when the reveal button is pressed?

Reply:
Great idea.